hamp_radar/iquick.py

```diff
@@ -366,6 +366,47 @@ def decode_iq(rawdata):
 }
 
 
+def single_dspparams_data(data):
+    """
+    Returns raw arrays of data from the first occurrence of DSP parameters
+    up to the next occurrence or up to the end of the data.
+
+    The first value in the returned list of raw arrays is the DSP parameters
+    configuration (tag == "PPAR").
+
+    Raises warning if no PPiAR tags are found in the data.
+
+    Parameters:
+    data (any): The input data (memory map or open file) from which to extract the
+                arrays associated with a single DSP parameter configuration.
+
+    Returns:
+    list: A list of raw arrays extracted from the data.
+    """
+
+    mmbgs = get_geometry(data)
+
+    start = None
+    end = -1
+    for i, mmbg in enumerate(mmbgs):
+        if mmbg.tag == "PPAR":
+            if start is None:
+                start = i
+            else:
+                end = i
```

Comment:
I'm not completely sure if this really does what it should do.
I guess the start case (e.g. start of file or first PPAR) should be fine. However, if there would be a case where multiple consecutive chain-type-2 (e.g. SRVI etc...) multi main blocks would be found (it's allowed but I guess it should normally not happen), then the resulting dataset might also be a bit oddly defined...

Reply:
Yes I agree with you but that would also be a very strange case. Also in the current writing of the dataset this wouldn't cause an issue since PPARs are ignored. Of course that's something we should take care of if/when we change how the datasets are created.

For now I think it's okay though - you too or you have other ideas?

Reply:
Yes, it really shouldn't happen, otherwise the data acquisition would be programmed in a strange way. But according to the docs, it would be totally valid.